synapse/handlers/receipts.py

```diff
@@ -165,43 +165,32 @@ def __init__(self, hs: "HomeServer"):
         self.config = hs.config
 
     @staticmethod
-    def filter_out_private(events: List[JsonDict], user_id: str) -> List[JsonDict]:
+    def filter_out_private(events: List[JsonDict], our_user_id: str) -> List[JsonDict]:
         """
         This method takes in what is returned by
         get_linearized_receipts_for_rooms() and goes through read receipts
         filtering out m.read.private receipts if they were not sent by the
         current user.
         """
 
-        visible_events = []
-
+        events = events.copy()
```

Comment:
This unfortunately won't make a deep-copy, so the deletion below will still mutate the cached value. 😢